File: 1470-shuffle-the-array/1470-shuffle-the-array.js
/**
 * @param {number[]} nums
 * @param {number} n
 * @return {number[]}
 */
var shuffle = function(nums, n) {
    var shuffled = []
    for (var i = 0; i < n; i++) {shuffled.push(nums[i],nums[n + i])}
    return shuffled
};

File: 1470-shuffle-the-array/1470-shuffle-the-array.rb
# @param {Integer[]} nums
# @param {Integer} n
# @return {Integer[]}
def shuffle(nums, n)
    
    left = 0
    right = nums.length / 2
    temp = right
    result = []
    while left < temp do
        result.push(nums[left], nums[right])
        left+=1
        right+=1
    end
    result
end

#  [2,5,1,3,4,7]
#   x x x y y y
#.  2,3,5,4,1,7